Zend/zend_weakrefs.c

```diff
@@ -597,11 +597,7 @@ ZEND_METHOD(WeakMap, getIterator)
 
 void zend_register_weakref_ce(void) /* {{{ */
 {
-	zend_class_entry ce;
-
-	INIT_CLASS_ENTRY(ce, "WeakReference", class_WeakReference_methods);
-	zend_ce_weakref = zend_register_internal_class(&ce);
-	zend_ce_weakref->ce_flags |= ZEND_ACC_FINAL | ZEND_ACC_NO_DYNAMIC_PROPERTIES;
+	zend_ce_weakref = register_class_WeakReference();
 
 	zend_ce_weakref->create_object = zend_weakref_new;
 	zend_ce_weakref->serialize = zend_class_serialize_deny;
```

Comment:
I'm seeing a lot of zend_class_serialize_deny/unserialize_deny in internal classes - would `/** @serialize-deny */` be useful to add

Reply:
Yes, I was also considering doing so, but I'm still not sure about it. If static analyzers would use these annotations, then yes, it could be useful to expose this information.

But then we could also expose if an object can't be instantiated directly. E.g. `CurlMultiHandle` has the following constructor handler:

```
static zend_function *curl_multi_get_constructor(zend_object *object) {
	zend_throw_error(NULL, "Cannot directly construct CurlMultiHandle, use curl_multi_init() instead");
	return NULL;
}
```

I think all of this could be generated by adding the following annotation:
```
@constructor-deny Cannot directly construct CurlMultiHandle, use curl_multi_init() instead`)
```

What do you guys think? If you think these features are worth to add, then I'll implement them.

Reply:
Not sure about this. Currently we don't assign any class handlers, so that provides a clear separation for what is and isn't generated.

Reply:
> Not sure about this. Currently we don't assign any class handlers, so that provides a clear separation for what is and isn't generated.

Yes, agreed!